### ISSUE TITLE
Gollum 5.1.2 displays Chinese path name as URI

### ISSUE THREAD

**Issue by fanhongtao** [gollum/gollum#1658]
Some of my wiki pages are in a directory named `数学`,  which means `maths` in English.

* In Gollum 5.1, the path is displayed as `数学`
  ![gollum-5 1](https://user-images.githubusercontent.com/402665/105158244-bbb66d80-5b48-11eb-9bbd-105942725193.png)

* In Gollum 5.1.2, the path is displayed as `%E6%95%B0%E5%AD%A6`
   ![gollum-5 1 2](https://user-images.githubusercontent.com/402665/105158521-046e2680-5b49-11eb-8d5e-06a9b41e96e9.png)

After search in the git logs, I find it was introduced by  [Guard against malicious file names](https://github.com/gollum/gollum/commit/137728cdabc0f60859fcd30404ad2b8fff6ef715#diff-9a4bb08f6de673dd18e80f29cd9ed26211e3ba56ace35db0210b3cc9dafcc4e7)

Obviously, we can not just use `CGI.unescape` to solve the problem.

---

Test code:

```ruby
#!/usr/bin/ruby

require 'cgi'
puts CGI.escape("数学")
```

**Comment by bartkamphorst** [comment]
Thanks for the report @fanhongtao . Any ideas for a fix?

**Comment by fanhongtao** [comment]
@bartkamphorst Maybe we can have a limitation, some characters can not be used in path/file names.

Or we can add an option, so the user can choose:
 1. Use arbitary characters in names & display names as URI
 2. Use limited characters in names & display the original names

**Comment by dometto** [comment]
I think rather than calling `CGI.escape` we should be using https://github.com/flavorjones/loofah (already part of gollum) to guard against malicious file names.

**Comment by dometto** [comment]
By the way: this is my bad. Sorry! I wanted to merge  a fix guarding against the vulnerability ASAP and thus didn't stop to think about whether `CGI.escape` was the right solution.

**Comment by benjaminwil** [comment]
I wonder if just using `CGI#escapeHTML` is acceptable?

```
[5] pry(main)> CGI.escape "数学"
=> "%E6%95%B0%E5%AD%A6"
[6] pry(main)> CGI.escapeHTML "数学"
=> "数学"
[7] pry(main)> CGI.escapeHTML '<img src="malicious-content" alt="数学" />'
=> "&lt;img src=&quot;malicious-content&quot; alt=&quot;数学&quot; /&gt;"
```

**Comment by dometto** [comment]
@benjaminwil good idea! Probably much faster than a call to `loofah`. If anyone cares to open a PR I'll be happy to merge it ASAP. Otherwise it may be a couple of weeks before I can properly work on gollum myself.

**Comment by benjaminwil** [comment]
I've opened a PR, #1663.

I looked into the benefits of using `loofah`'s helpers over the built-in `CGI#escapeHTML` method, and it wasn't immediately obvious to me. So I went with the built-in method.

Generally, breadcrumb items shouldn't contain HTML anyway. Where `loofah` seems to shine is when sanitizing user input that is more complex and contains HTML or XML. So sanitizing the breadcrumbs with `#escapeHTML` seemed like an okay choice here.
